sandbox/cgroup/cgroup.go

```diff
@@ -263,18 +263,21 @@ func ProcessPathInTrackingCgroup(pid int) (string, error) {
 	// It will forever stay present in the kernel but will not be present in
 	// the file-system. As such, allow v2 to register only if it is really
 	// mounted on the system.
-	var allowV2 bool
+	var useV2 bool
 	if ver, err := Version(); err != nil {
 		return "", err
 	} else if ver == V2 {
-		allowV2 = true
+		useV2 = true
 	}
 	entry, err := scanProcCgroupFile(fname, func(e *procInfoEntry) bool {
-		if e.CgroupID == 0 && allowV2 {
-			return true
-		}
-		if len(e.Controllers) == 1 && e.Controllers[0] == "name=systemd" {
```

Comment:
looking at your description of the issue another approach would have been to change this line to also include:

```
&& e.Path != "/" 
```

?

Reply:
No, because we are not sure that it will be `/`: in my tests, it looks like systemd will place our process in the deepest matching existing cgroup: I've seen the process placed in a subdirectory of `/user.slice/user-1000.slice/`. In other words, it looks like new subdirectories cannot be created, but existing files can be written to. This [snippet in systemd](https://github.com/systemd/systemd/blob/e3a4724db2fca3e0269e44dccb4c3ec692fb1943/src/shared/cgroup-setup.c#L384-L393) seems to confirm that the logic is that systemd will try to place the process in the deepest existing cgroup.